src/app/components/translate-attributes/translate-attributes.component.ts

```diff
@@ -0,0 +1,341 @@
+        } catch (err) {
+          this.handleError(err.message);
+        }
+      });
+  }
+
+  mulitipleActionSortPromises(): void {
+    // Sort promises
+    // 1. delete items first ()
+    // 2. update items second
+    // 3. create items last
+    this.multipleActionsPromises.sort((a, b) => {
+      if (b.type === 'delete') {
+        return 1;
+      }
+      if (a.type === 'delete') {
+        return -1;
+      }
+      if (a.type === 'create' && b.type === 'create') {
+        return -1;
+      }
+      if (a.type === 'create') {
+        return 1;
+      }
+      if (b.type === 'create') {
+        return -1;
+      }
+    });
+  }
+
+  async mulitipleActionDelete(promise: IPromises): Promise<Boolean> {
+    console.log('mulitipleActionDelete', 'Start');
+    await this.attributeTranslationService
+      .delete(promise.data)
+      .then(() => {
+        this.multipleActionsResults.push({
+          type: 'success',
+          text: `Successfully deleted attribute ${promise.id}`,
+        });
+      })
+      .catch(() => {
+        this.multipleActionsError = true;
+
+        this.multipleActionsResults.push({
+          type: 'danger',
+          text: `Error while deleting attribute ${promise.id}`,
+        });
+      });
+    console.log('mulitipleActionDelete', 'End');
+    return true;
+  }
+
+  async mulitipleActionUpdate(promise: IPromises): Promise<Boolean> {
+    console.log('mulitipleActionUpdate', 'Start');
+    await this.attributeTranslationService
+      .update(promise.data)
+      .then(() => {
+        this.multipleActionsResults.push({
+          type: 'success',
+          text: `Successfully updated attribute ${promise.id}`,
+        });
+      })
+      .catch(() => {
+        this.multipleActionsError = true;
+
+        this.multipleActionsResults.push({
+          type: 'danger',
+          text: `Error while updating attribute ${promise.id}`,
+        });
+      });
+    console.log('mulitipleActionUpdate', 'End');
+    return true;
+  }
+
+  async mulitipleActionCreate(promise: IPromises): Promise<Boolean> {
+    console.log('mulitipleActionCreate', 'Start');
+    await this.attributeTranslationService
+      .create(this.resource.id, promise.data)
+      .then(() => {
+        this.multipleActionsResults.push({
+          type: 'success',
+          text: `Successfully created attribute ${promise.id}`,
+        });
+      })
+      .catch(() => {
+        this.multipleActionsError = true;
+
+        this.multipleActionsResults.push({
+          type: 'danger',
+          text: `Error while creating attribute ${promise.id}`,
+        });
+      });
+    console.log('mulitipleActionCreate', 'End');
+    return true;
+  }
+
+  createAttribute(): void {
+    // CHECK 1: Ensure Keys aren't the same
+    const generateId = Math.floor(Math.random() * 999999);
+    console.log('generateId', generateId);
+    this.changeMade();
+
+    const attribute: AttributeTranslation = {
+      id: generateId,
```

Comment:
I'm curious what the need was for generating a random id when creating an attribute?

Reply:
What I've done in the past when generating a random id for new objects is generate a random negative number because that will never conflict with an auto-incremented id.

Reply:
negative ids also provide a quick `if(id < 0)` check that determines if it's a new object

Reply:
We need the `ID` when we call `checkRemoteResourceForDifferences` as I use JS Map (JS version of hashmap); I use `ID` as the key for each Map entry.